applications/StructuralMechanicsApplication/custom_strategies/custom_strategies/eigensolver_strategy.hpp

```diff
@@ -668,33 +679,33 @@ class EigensolverStrategy
         // store eigenvalues in process info
         rModelPart.GetProcessInfo()[EIGENVALUE_VECTOR] = rEigenvalues;
 
-        for (ModelPart::NodeIterator itNode = rModelPart.NodesBegin(); itNode!= rModelPart.NodesEnd(); itNode++)
-        {
+        const auto& r_dof_set = this->pGetBuilderAndSolver()->GetDofSet();
+
+        for (ModelPart::NodeIterator itNode = rModelPart.NodesBegin(); itNode!= rModelPart.NodesEnd(); itNode++) {
             ModelPart::NodeType::DofsContainerType& NodeDofs = itNode->GetDofs();
             const std::size_t NumNodeDofs = NodeDofs.size();
             Matrix& rNodeEigenvectors = itNode->GetValue(EIGENVECTOR_MATRIX);
-            if (rNodeEigenvectors.size1() != NumEigenvalues || rNodeEigenvectors.size2() != NumNodeDofs)
-            {
+            if (rNodeEigenvectors.size1() != NumEigenvalues || rNodeEigenvectors.size2() != NumNodeDofs) {
                 rNodeEigenvectors.resize(NumEigenvalues,NumNodeDofs,false);
             }
 
-            // TO BE VERIFIED!! In the current implmentation of Dofs there are nor reordered and only pushec back. 
-            // // the jth column index of EIGENVECTOR_MATRIX corresponds to the jth nodal dof. therefore,
-            // // the dof ordering must not change.
-            // if (NodeDofs.IsSorted() == false)
-            // {
-            //     NodeDofs.Sort();
-            // }
-
             // fill the EIGENVECTOR_MATRIX
-            for (std::size_t i = 0; i < NumEigenvalues; i++)
+            for (std::size_t i = 0; i < NumEigenvalues; i++) {
                 for (std::size_t j = 0; j < NumNodeDofs; j++)
                 {
-                    auto itDof = std::begin(NodeDofs) + j;
-                    rNodeEigenvectors(i,j) = rEigenvectors(i,(*itDof)->EquationId());
+                    const auto itDof = std::begin(NodeDofs) + j;
+                    bool is_active = !(r_dof_set.find(**itDof) == r_dof_set.end());
```

Comment:
I checked whether this has an effect on performance
In a Model with 500K Nodes it had no measureable impact